src/EventStore.php

```diff
@@ -206,10 +209,16 @@ public function writeToStream($streamName, WritableToStream $events, $expectedVe
         $this->sendRequest($request);
 
         $responseStatusCode = $this->getLastResponse()->getStatusCode();
-
         if (ResponseCode::HTTP_BAD_REQUEST == $responseStatusCode) {
             throw new WrongExpectedVersionException();
         }
+
+        $locationHeaders = $this->getLastResponse()->getHeader('Location');
+        if (!empty($locationHeaders[0]) && strpos($locationHeaders[0], $streamUrl) === 0) {
+            return (int)trim(substr($locationHeaders[0], strlen($streamUrl)), '/');
+        }
```

Comment:
I think that it's better to extract these 4 lines in a private method with a meaningful name